cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/sql/GcpDatabaseServerUpgradeService.java

```
@@ -1,21 +1,96 @@
+        String availabilityZone = ac.getCloudContext().getLocation().getAvailabilityZone().value();
+        List<CloudResource> buildableResource = new ArrayList<>();
+        buildableResource.add(getGcpDatabase(deploymentName, availabilityZone));
+
+        try {
+            InstancesListResponse list = sqlAdmin.instances().list(projectId).execute();
+            Optional<DatabaseInstance> first = Optional.empty();
+            if (!list.isEmpty()) {
+                first = list.getItems()
+                        .stream()
+                        .filter(e -> e.getName().equals(deploymentName))
+                        .findFirst();
+            }
+            if (!first.isEmpty()) {
+                DatabaseInstance databaseInstance = first.get();
+                databaseInstance.setDatabaseVersion(databaseServerView.getDatabaseType() + "_" + databaseVersion.getVersion());
+                SQLAdmin.Instances.Patch patch = sqlAdmin.instances().patch(projectId, databaseInstance.getName(), databaseInstance);
+                patch.setPrettyPrint(Boolean.TRUE);
+                try {
+                    Operation operation = patch.execute();
+                    verifyOperation(operation, buildableResource);
+                    CloudResource operationAwareCloudResource = createOperationAwareCloudResource(buildableResource.get(0), operation);
+                    databasePollerService.upgradeDatabasePoller(ac, List.of(operationAwareCloudResource));
+                    buildableResource.forEach(dbr -> resourceNotifier.notifyUpdate(dbr, ac.getCloudContext()));
+                    return Collections.singletonList(operationAwareCloudResource);
+                } catch (GoogleJsonResponseException e) {
+                    throw new GcpResourceException(checkException(e), resourceType(), buildableResource.get(0).getName());
+                }
+            } else {
+                LOGGER.debug("Deployment does not exist: {}", deploymentName);
```

this should notify the caller with an exception and cause the upgrade flow to fail, doesn't it?

---

I would throw an exception to inform the user that upgrade was not possible due to this.

---

Fixed: will throw GcpResourceException if deployment does not exist.